### PR TITLE
LPS-98143

### PR DESCRIPTION
**Pull request by tinatian**
Hi @topolik ,

This pull is to improve performance for startup, I am sending it to you cause it has some changes I am not 100% sure, so would like you to do a double check.

cc @shuyangzhou

Thanks.
Tina.

### CODE REVIEW

**Comment by tinatian** [comment]
ci:test:relevant

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/LiferayAccessTokenService.java [hunk 1]
```diff
@@ -88,4 +89,10 @@ protected Client authenticateClientIfNeeded(
 		return client;
 	}
 
+	@Override
+	protected void injectContextIntoOAuthProviders() {
+		OAuthUtils.injectContextIntoOAuthProvider(
+			getMessageContext(), getDataProvider());
+	}
+
```
Comment (topolik) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/LiferayAccessTokenService.java]: Hi @tinatian, our LiferayOAuthDataProvider also doesn't define setMessageContext() so we could skip it for this too?

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/LiferayAccessTokenServiceRegistrator.java [hunk 2]
```diff
@@ -131,8 +111,14 @@ private void _updateLiferayAccessTokenService(BundleContext bundleContext) {
 			liferayAccessTokenServiceProperties);
 	}
 
-	private final List<AccessTokenGrantHandler> _accessTokenGrantHandlers =
-		new ArrayList<>();
+	@Reference(
+		cardinality = ReferenceCardinality.AT_LEAST_ONE,
+		fieldOption = FieldOption.UPDATE, policy = ReferencePolicy.DYNAMIC,
```
Comment (topolik) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/LiferayAccessTokenServiceRegistrator.java]: Nice: `fieldOption = FieldOption.UPDATE`

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/LiferayAccessTokenService.java (hunk 1 above)
Comment (tinatian) [reply modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/LiferayAccessTokenService.java]: @topolik Yes, I think so, then we just need an empty method to override the parent logic here.

**Comment by topolik** [comment]
Hi @tinatian it looks fine. I'm going to holidays, may I ask you to resend @csierra considering the https://github.com/topolik/liferay-portal/pull/792#pullrequestreview-267299423? Thank you!

**Comment by tinatian** [comment]
@topolik Sure, I will. Thanks for reviewing! Have a good holiday!

Tina.

**Comment by topolik** [comment]
Closing, already merged. Thank you.

**Pull request closed without merging**